src/types.ts

```diff
@@ -99,10 +117,21 @@ export interface StreamUpdate {
   fetching: boolean;
 }
 
+/** A new response/update from the [client]{@link Client} subscription stream. */
+export interface SubscriptionStreamUpdate {
+  /** Any data returned from a [GraphQL]{@link GraphQL} query. */
+  data?: ExchangeResult['data'];
+  /** Any errors returned from a [GraphQL]{@link GraphQL} query. */
+  error?: ExchangeResult['error'];
+}
+
 /** Arguments for creating a client instance. */
 export interface CreateClientInstanceOpts {
   /** A callback function for when changes occur. */
   onChange: (data: StreamUpdate) => any;
+
+  /** A callback function for when subscriptions return new data. */
+  onSubscriptionUpdate: (data: SubscriptionStreamUpdate) => any;
```

Comment:
Is there any way of determining which subscription the data is being returned from?

I think it would be simpler to just pass a subscription back from the `executeSubscription` function?

```tsx
client.executeSubscription(subscription).subscribe((data) => console.log(`subscription ${subscription} returned data ${data}`))
```

Reply:
I could, but I was trying to follow the general pattern of the library. There may be downsides to this approach where each call site needs to manage the state. Whereas when the client handles it it's centralized.

Determing which subscription is actually easy because of the shape of the data.

`const subscriptionName = Object.keys(data)[0]`